src/Bridge/Doctrine/Common/Extension/FilterExtension.php

```diff
@@ -47,7 +46,7 @@ public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFa
     /**
      * {@inheritdoc}
      */
-    public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass = null, string $operationName = null, array $context = [])
+    public function applyToCollection($builder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass = null, string $operationName = null, array $context = [])
```

Comment:
Couldn't we have two interfaces to describe two different kind of filters (one for mongodb, another for sql)?